### ISSUE TITLE
Karpenter roles to follow new policy guide from new release v0.32.x

### ISSUE THREAD

**Issue by guilinden** [terraform-aws-modules/terraform-aws-eks#2809]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - No 🛑: please wait to file a request until the functionality is avaialble in the AWS provider
  - Yes ✅: please list the AWS provider version which introduced this functionality

## Is your request related to a problem? Please describe.
The new Karpenter version moved the product to beta and changed how the IAM policies for the controller and nodes should be configured. 

## Describe the solution you'd like.
Follow Karpenter provided policy: https://github.com/aws/karpenter/blob/main/website/content/en/docs/upgrading/v1beta1-controller-policy.json

## Describe alternatives you've considered.
You could provide a new additional policy, but I believe the default behavior of the module should be aligned with the latest Karpenter version

## Additional context
I would like to work on this feature if there is no active work yet

**Comment by johnjeffers** [comment]
Is there any timeline for v20.0.0? EKS 1.29 has kinda forced my hand on dealing with Karpenter.

```
ERROR controller.awsnodetemplate karpenter version is not compatible with K8s version 1.29 
```

Does anyone know of any workarounds for dealing with the karpenter 32.0+ changes in the meantime, short of abandoning the module and managing it myself?

**Comment by bryantbiggs** [comment]
@johnjeffers thats not related to this module - v20 should be live today/tomorrow and it has an IAM policy aligned with the current upstream project recommendation but you will still see that error message due to https://github.com/aws/karpenter-provider-aws/issues/5528

**Comment by johnjeffers** [comment]
@bryantbiggs Thanks for that info. I know that the module isn't the reason for the error message. What I didn't know is that Karpenter 0.33 isn't compatible with k8s 1.29. I thought I'd be able to get past that error by upgrading, which I hadn't done yet because I was waiting for v20 of the module.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.0.0 :tada:
